a3fe/tests/test_run_integration.py

```diff
@@ -0,0 +1,214 @@
+            job_path = os.path.join(calc.base_dir, "input", jobname)
+            assert os.path.exists(job_path)
+
+        # Check that the required leg directories exist
+        for leg in LEGS_WITH_STAGES.keys():
+            leg_dir = os.path.join(calc.base_dir, leg)
+            assert os.path.exists(leg_dir)
+
+            # Check that the required stage directories exist
+            for stage in LEGS_WITH_STAGES[leg]:
+                stage_dir = os.path.join(leg_dir, stage)
+                assert os.path.exists(stage_dir)
+                output_dir = os.path.join(stage_dir, "output")
+                assert os.path.exists(output_dir)
+
+    @pytest.mark.integration
+    def test_slurm_non_adaptive_run(self, slurm_calc, system_prep_config):
+        """
+        Full integration test for non-adaptive run using SLURM.
+        """
+        calc = self._setup_calculation(slurm_calc, system_prep_config)
+
+        # Run a short non-adaptive calculation
+        calc.run(adaptive=False, runtime=0.1)
+        calc.wait()
+
+        # Check that the calculation is not running
+        assert not calc.running
+
+        # Set the entire simulation time to equilibrated for analysis
+        for leg in calc.legs:
+            for stage in leg.stages:
+                for lam_win in stage.lam_windows:
+                    lam_win._equilibrated = True
+                    lam_win._equil_time = 0
+
+        # Check that the simulation time for each lambda window is correct
+        for leg in calc.legs:
+            for stage in leg.stages:
+                for lam_win in stage.lam_windows:
+                    assert round(lam_win.tot_simtime / lam_win.ensemble_size, 1) == 0.1
+
+        # Run analysis with error handling
+        try:
+            calc.analyse()
+
+            # Check that the output file exists
+            assert os.path.exists(
+                os.path.join(calc.base_dir, "output", "overall_stats.dat")
+            )
+
+            # Check that the free energy change is reasonable
+            with open(
+                os.path.join(calc.base_dir, "output", "overall_stats.dat"), "r"
+            ) as f:
+                try:
+                    dg = float(f.readlines()[1].split(" ")[3])
+                    assert dg < -5
+                    assert dg > -25
+                except (IndexError, ValueError) as e:
+                    pytest.fail(f"Failed to parse free energy value: {str(e)}")
+
+        except FileNotFoundError as e:
+            import warnings
+
+            warnings.warn(f"SLURM job file not found: {str(e)}")
+
+    @pytest.mark.integration
+    def test_slurm_optimal_lambda(self, slurm_calc, system_prep_config):
+        """Test that the optimal lambda window creation function works."""
+        calc = self._setup_calculation(slurm_calc, system_prep_config)
+
+        calc.get_optimal_lam_vals(delta_er=2)
+
+        # Check that the old runs have been moved, and that the new lambda values are
```

Comment:
Please could you check that the new lambda values are different to the old ones? Thanks!